### PR TITLE
Clamp controller sticks to circle, instead of square

### PR DESCRIPTION
**Pull request by emmauss**
Xinput devices provides axis data with square ranges, ie +-32767. Moving the sticks to diagonal ends means that it will have the max range values for both axes. This causes fast motion of player characters in games that do not clamp their input. 

This fixes it by clamping the input to a circle. Fixes Link sprinting without using up stamina when the controller sticks are at diagonals

### CODE REVIEW

File: Ryujinx.Input/HLE/NpadController.cs [hunk 1]
```diff
@@ -3,6 +3,7 @@
 using Ryujinx.Common.Configuration.Hid.Controller;
 using Ryujinx.Common.Configuration.Hid.Controller.Motion;
 using Ryujinx.HLE.HOS.Services.Hid;
+using SixLabors.ImageSharp;
```
Comment (AcK77) [inline Ryujinx.Input/HLE/NpadController.cs:6]: Is it really needed? If yes, could it not be replaced with something else?

File: Ryujinx.Input/HLE/NpadController.cs [hunk 2]
```diff
@@ -385,16 +364,57 @@ public GamepadInput GetHLEInputState()
             }
             else if (_config is StandardControllerInputConfig controllerConfig)
             {
-                (float leftAxisX, float leftAxisY) = State.GetStick(StickInputId.Left);
+                (float leftAxisX,  float leftAxisY)  = State.GetStick(StickInputId.Left);
                 (float rightAxisX, float rightAxisY) = State.GetStick(StickInputId.Right);
 
-                state.LStick = ApplyDeadzone(leftAxisX, leftAxisY, controllerConfig.DeadzoneLeft);
-                state.RStick = ApplyDeadzone(rightAxisX, rightAxisY, controllerConfig.DeadzoneRight);
+                state.LStick = ClampToCircle(ApplyDeadzone(leftAxisX, leftAxisY,  controllerConfig.DeadzoneLeft));
```
Comment (AcK77) [inline Ryujinx.Input/HLE/NpadController.cs:370]:
```suggestion
                state.LStick = ClampToCircle(ApplyDeadzone(leftAxisX, leftAxisY, controllerConfig.DeadzoneLeft));
```

File: Ryujinx.Input/HLE/NpadController.cs [hunk 3]
```diff
@@ -385,16 +364,57 @@ public GamepadInput GetHLEInputState()
...
+            if (value <= -short.MaxValue)
+            {
+                return -short.MaxValue;
+            }
+            else
+            {
+                return (short)(value * short.MaxValue);
+            }
+        }
+
```
Comment (AcK77) [inline Ryujinx.Input/HLE/NpadController.cs:399]: Remove extra empty line

**Review by AcK77 — CHANGES_REQUESTED** [review]

File: Ryujinx.Input/HLE/NpadController.cs (hunk 3 above)
Comment (emmauss) [reply Ryujinx.Input/HLE/NpadController.cs:398]: addressed

File: Ryujinx.Input/HLE/NpadController.cs (hunk 2 above)
Comment (emmauss) [reply Ryujinx.Input/HLE/NpadController.cs]: addressed

File: Ryujinx.Input/HLE/NpadController.cs (hunk 1 above)
Comment (emmauss) [reply Ryujinx.Input/HLE/NpadController.cs]: I don't know how that got there. removed

**Review by AcK77 — APPROVED** [review]
Lgtm! Thanks!

**Comment by MelonSpeedruns** [comment]
LGTM, tested with Skyward Sword HD, and no super-walking when the joystick is diagonal!

**Review by Thog — APPROVED** [review]

**Pull request merged**